Sources/Common/MultipartRouteType.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  MultipartRouteType.swift
+//  OperaSwift
+//
+//  Created by Miguel Revetria on 5/3/17.
+//
+//
+
+import Alamofire
+import Foundation
+import UIKit
+
+// MARK: - MultipartRequestType
+
+public struct MultipartData {
+
+    let data: Data
+    let fileName: String
+    let mimeType: String
+    let name: String
+
+}
+
+public protocol MultipartRouteType: RouteType {
+
+    var items: [MultipartData] { get }
+
+}
+
+public extension MultipartRouteType {
+
+    var method: HTTPMethod {
+        return .post
+    }
+
+}
+
+// MARK: - ImageUploadRouteType
+
+public enum ImageUploadEncoding {
+
+    case png
+    case jpeg(quality: CGFloat)
+
+    var fileExtension: String {
+        switch self {
+        case .png:
+            return "png"
+        case .jpeg:
+            return "jpg"
+        }
+    }
+
+    func encode(image: UIImage) -> Data? {
+        switch self {
+        case .png:
+            return UIImagePNGRepresentation(image)
+        case let .jpeg(quality):
+            return UIImageJPEGRepresentation(image, quality)
+        }
+    }
+
+}
+
+public protocol ImageUploadRouteType: MultipartRouteType {
+
+    var encoding: ImageUploadEncoding { get }
+    var image: UIImage { get set }
+    var fileName: String { get }
+    var imageName: String { get }
+    var mimeType: String { get }
+
```

Comment:
I'm not sure about letting the user be creative with `mimeType` and `fileName`. For example for the `mimeType`, if the encoding is `jpg` the user can provide the `mimeType` as `image/png` by mistake.

Maybe `mimeType` and `fileName` must be only `extension` variables and not declared in the protocol. Not sure about the `fileName` though...